### PR TITLE
Bump to 1.2.5 + migrate to osx_arm64

### PR DESCRIPTION
**Pull request by hadim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hadim** [comment]
close #6 and #7 

I also added myself as a maintainer and can take care of the py312 + boost migration.

**Comment by hadim** [comment]
@simonbray @h-vetinari ready for review

File: recipe/vina.patch
Comment (h-vetinari) [inline recipe/vina.patch:152]: That shouldn't be necessary. Also, if `locate_boost` is used anywhere but during `setup.py`, it will give the wrong result (`PREFIX` is populated only during the build; `CONDA_PREFIX` also in the final environment).

File: recipe/vina.patch
Comment (h-vetinari) [inline recipe/vina.patch:123]: Please avoid these spurious changes in the patch

File: recipe/vina.patch
Comment (h-vetinari) [inline recipe/vina.patch:4]: Please keep the patch format; generate it as `git format-patch <tag_for_1.2.5> --no-signature`

File: recipe/recipe-scripts-license.txt
Comment (h-vetinari) [inline recipe/recipe-scripts-license.txt:1]: What's the intention of this file? We don't seem to use this anywhere else, with the exception of a [handful](https://github.com/search?q=org%3Aconda-forge%20path%3Arecipe-scripts-license.txt&type=code) of feedstocks where you added the same.

File: recipe/vina.patch
Comment (h-vetinari) [inline recipe/vina.patch:54]: I'd prefer not to even patch the `BOOST_STATIC=y` case. It should just fail, because we want to build against the shared libs (or just the headers, if that suffices).

**Review by h-vetinari — COMMENTED** [review]
Thanks for fixing this. Some relatively minor issues, then we should be good to go!

File: recipe/recipe-scripts-license.txt
Comment (hadim) [reply recipe/recipe-scripts-license.txt:1]: they got added automatically - so i assumed those were needed to be committed. will remove

File: recipe/vina.patch
Comment (hadim) [reply recipe/vina.patch:123]: those are added automatically upon saving the file - it's not super practical to remove those but i disabled the option for now.

File: recipe/vina.patch
Comment (hadim) [reply recipe/vina.patch]: Boost is not found without it. I will put it back.

Comment (hadim) [reply recipe/vina.patch:114]: Let me know if you have another idea - but using PREFIX seems to work.

**Comment by hadim** [comment]
@h-vetinari LGTM

**Comment by hadim** [comment]
@h-vetinari is there, is anything else you want me to modify here?

File: recipe/vina.patch
Comment (h-vetinari) [reply recipe/vina.patch:126]: It's OK, it seems to only happen at build time.

**Review by h-vetinari — APPROVED** [review]
> @h-vetinari is there, is anything else you want me to modify here?

I'm not the maintainer here - ideally @simonbray should approve. If he's not available, I can merge it.

**Comment by hadim** [comment]
Thanks @h-vetinari .

Let's wait a bit if we hear from Simon.

**Comment by hadim** [comment]
@h-vetinari ok to merge here?

**Comment by hadim** [comment]
I also tested boost 1.82 locally and it seems all good. Will trigger the boost migrations again once this one is merged.

**Review by simonbray — APPROVED** [review]

**Comment by simonbray** [comment]
Sorry for the slow response - I'm afraid I've mostly tuned-out conda-forge things these days. Thanks @hadim for helping with keeping the recipe up-to-date.

**Comment by hadim** [comment]
Thanks @simonbray !

**Pull request merged**
